File: python_peer2/start.py
from file_server import FileServer
from main import main  #menu based client
from advertise import advertise_service, stop_advertisement
import sys

def start():
    print("Welcome to P2P File Share")
    print("1. Receive a file")
    print("2. Send a file (standby to recieve file request)")
    choice = input("Select your role (1 or 2): ").strip()

    if choice == "1":
        print("📤 Starting in reciever mode...")
        main()
        
    elif choice == "2":
        print("📥 Starting in send mode...")
        service_name = advertise_service()
        server = FileServer()
        print("👋 Press Ctrl+C to stop the server at any time.")

        try:
            server.start()  # runs in the foreground so input() works
        except KeyboardInterrupt:
            print("\n🛑 Shutting down...")
            stop_advertisement()
            sys.exit(0)
        else:
            print("❌ Invalid selection.")
            sys.exit(1)

if __name__ == "__main__":
    start()

File: ruby_peer2/start.rb
require_relative "file_server"
require_relative "advertise"
require_relative "main" # <- your interactive peer client

def start
  puts "Welcome to P2P File Share"
  puts "1. Receive a file"
  puts "2. Send a file (standby to receive file request)"
  print "Select your role (1 or 2): "
  choice = gets.strip

  case choice
  when "1"
    puts "📤 Starting in receiver mode..."
    run_menu # defined in main.rb

  when "2"
    puts "📥 Starting in send mode..."
    advertise_service
    server = FileServer.new
    puts "👋 Press Ctrl+C to stop the server at any time."

    begin
      server.start
    rescue Interrupt
      puts "\n🛑 Shutting down..."
      stop_advertisement
      exit(0)
    end

  else
    puts "❌ Invalid option."
    exit(1)
  end
end

start if __FILE__ == $0
